src/item_use.c

```diff
@@ -1323,4 +1325,51 @@ void ItemUseOutOfBattle_CannotUse(u8 taskId)
     DisplayDadsAdviceCannotUseItemMessage(taskId, gTasks[taskId].tUsingRegisteredKeyItem);
 }
 
+void FieldUseFunc_VsSeeker(u8 taskId)
+{
+    u32 i;
+    u16 mapGroup = gSaveBlock1Ptr->location.mapGroup;
+    u16 mapNum = gSaveBlock1Ptr->location.mapNum;
+    bool32 isValidMap = (gMapHeader.mapType == MAP_TYPE_ROUTE ||
+            gMapHeader.mapType == MAP_TYPE_TOWN ||
+            gMapHeader.mapType == MAP_TYPE_CITY ||
+            gMapHeader.mapType == MAP_TYPE_OCEAN_ROUTE);
+
+    if (gMapHeader.mapType == MAP_TYPE_INDOOR) {
+        isValidMap = isValidMap || (mapGroup == MAP_GROUP(MT_PYRE_SUMMIT)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_NORTH)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_NORTHEAST)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_NORTHWEST)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_SOUTH)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_SOUTHEAST)) ||
+            (mapGroup == MAP_GROUP(SAFARI_ZONE_SOUTHWEST)) ||
+            (mapGroup == MAP_GROUP(SKY_PILLAR_TOP)) ||
+            (mapGroup == MAP_GROUP(SOUTHERN_ISLAND_EXTERIOR)) ||
+            (mapGroup == MAP_GROUP(SOUTHERN_ISLAND_INTERIOR)) ||
+            (mapGroup == MAP_GROUP(RUSTBORO_CITY_GYM)) ||
+            (mapGroup == MAP_GROUP(DEWFORD_TOWN_GYM)) ||
+            (mapGroup == MAP_GROUP(MAUVILLE_CITY_GYM)) ||
+            (mapGroup == MAP_GROUP(LAVARIDGE_TOWN_GYM_1F)) ||
+            (mapGroup == MAP_GROUP(LAVARIDGE_TOWN_GYM_B1F)) ||
+            (mapGroup == MAP_GROUP(PETALBURG_CITY_GYM)) ||
+            (mapGroup == MAP_GROUP(FORTREE_CITY_GYM)) ||
+            (mapGroup == MAP_GROUP(MOSSDEEP_CITY_GYM)) ||
+            (mapGroup == MAP_GROUP(SOOTOPOLIS_CITY_GYM_1F)) ||
+            (mapGroup == MAP_GROUP(SOOTOPOLIS_CITY_GYM_B1F));
+    }
```

Comment:
surely this can be made more readable & maintainable by extracting it to a (potentially inlined) function with a linear search?